docs/docs/adding-forms.md

```diff
@@ -2,7 +2,169 @@
 title: Adding forms
 ---
 
-This is a stub. Help our community expand it.
+Gatsby is built on top of React. So anything that is possible with a React form is possible in Gatsby. Let's start with the following page.
 
-Please use the [Gatsby Style Guide](/docs/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+```JSX
+import React from "react"
+
+export default () => <div>Hello world!</div>
+```
+
+This gatsby page is a React component. When you want to create a form, you need to store the state of the form - what the user has entered. Convert your function (stateless) component to a class (stateful) component.
+
+```JSX
+import React from "react"
+
+export default class IndexPage extends React.Component {
+    render() {
+        return (
+            <div>Hello world!</div>
+        )
+    }
+}
+```
+
+Now that you have created a class component, you can add `state` to the component.
+
+```JSX
+import React from "react"
+
+export default class IndexPage extends React.Component {
+    state = {
+        firstName: '',
+        lastName: '',
+    }
+
```

Comment:
nit: but let's tighten up the spacing here!

Reply:
I cleaned it up!